File: Lab1/excercises/UDPClient.java
import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

static final int port = 8008;
static final String message = "Message from Client";
static final int receiveLength = 100;

void send(DatagramSocket socket, InetAddress address) throws IOException {
    var buffer = message.getBytes();

    DatagramPacket toSend = new DatagramPacket(
        buffer,
        buffer.length,
        address,
        port
    );

    socket.send(toSend);

    System.out.format(
        "Sent \"%s\" to %s:%d\n", 
        message, 
        toSend.getAddress(),
        port
    );
}

void receive(DatagramSocket socket) throws IOException {
    var buffer = new byte[receiveLength];

    DatagramPacket toReceive = new DatagramPacket(
        buffer,
        buffer.length
    );

    socket.receive(toReceive);

    var message = new String(toReceive.getData());

    System.out.format(
        "Received \"%s\" from %s\n", 
        message, 
        toReceive.getAddress()
    );
}

void main() throws Exception {
    var address = InetAddress.getByName("localhost");

    try (
        DatagramSocket socket = new DatagramSocket()
    ) {
        send(socket, address);
        receive(socket);
    }
}

File: Lab1/excercises/UDPClient.py
import socket as sockets
from socket import socket as Socket

ADDRESS = '127.0.0.1'
RECEIVE = 9009
SEND = 8008
RECEIVE_LENGTH = 100
MESSAGE = 'Żółta gęś'
NUMBER = 300
ENCODING = 'cp1250'


def as_bytes(message: str) -> bytes:
    return bytes(message, ENCODING)


def bytify(number: int) -> bytes:
    return number.to_bytes(4, 'little')


def as_str(buffer: bytes) -> str:
    return str(buffer, ENCODING)


def as_int(buffer: bytes) -> int:
    return int.from_bytes(buffer, 'little')


def send(socket: Socket) -> None:
    buffer = as_bytes(MESSAGE)

    socket.sendto(
        buffer,
        (ADDRESS, SEND)
    )

    message = as_str(buffer)

    print(f'Sent "{message}" to {ADDRESS}:{SEND}')


def send_number(socket: Socket) -> None:
    buffer = bytify(NUMBER)

    socket.sendto(
        buffer,
        (ADDRESS, SEND)
    )

    message = as_int(buffer)

    print(f'Sent "{message}" to {ADDRESS}:{SEND}')


def receive(socket: Socket) -> None:
    buffer, address = socket.recvfrom(RECEIVE_LENGTH)
    message = as_str(buffer)

    print(f'Received "{message}" from {address}')


def receive_number(socket: Socket) -> None:
    buffer, address = socket.recvfrom(RECEIVE_LENGTH)
    message = as_int(buffer)

    print(f'Received "{message}" from {address}')


if __name__ == '__main__':
    socket = Socket(sockets.AF_INET, sockets.SOCK_DGRAM)
    socket.bind(('', RECEIVE))

    # send(socket)
    # receive(socket)

    send_number(socket)
    receive_number(socket)
